### DISCUSSION TITLE
[Support]: /tmp/cache fills without emptying automatically

### DISCUSSION THREAD

**Discussion by Pinez83** [blakeblackshear/frigate#11067] [General Support] [answered]
### Describe the problem you are having

Hi everyone, I've noticed that tmp/cache tends to fill up without emptying itself. I've already increased the disk space a few times, but there seems to be a different problem.

I use frigate with portainer on proxmox
![Screenshot 2024-04-22 160310](https://github.com/blakeblackshear/frigate/assets/96543064/8751fd60-e1b0-4429-90ac-dfb880ee0ce6)

### Version

 0.13.2-6476F8A

### Frigate config file

```yaml
mqtt:
  host: 192.168.0.56
  user: pinez
  password: xxx
  topic_prefix: frigate
  client_id: frigate

go2rtc:
  streams:
    Giardino_sub:
      - rtsp://admin:xxx@192.168.0.77:554/h264Preview_01_sub
      - "ffmpeg:Giardino#audio=opus"
    Giardino_main:
      - rtsp://admin:xxx@192.168.0.77:554/h264Preview_01_main
      - "ffmpeg:Giardino#audio=opus"



cameras:
  Ingresso:
    ffmpeg:
      inputs:
        - path: rtsp://admin:xxx@192.168.0.65:554/live2.sdp
          roles:
            - detect
        - path: rtsp://admin:xxx@192.168.0.65/live1.sdp
          roles:
            - record
      hwaccel_args: -c:v h264_qsv
    rtmp:
      enabled: false
    detect:
      width: 480
      height: 272
      fps: 7
    motion:
      mask:
        - 265,87,311,92,319,55,269,51
    objects:
      track:
        - person
        - bicycle
        - dog
        - cat
        - car
    snapshots:
      enabled: true
      bounding_box: true
  Garage:
    ffmpeg:
      inputs:
        - path: rtsp://admin:xxx@192.168.0.209:554/cam/realmonitor?channel=1&subtype=1&unicast=true&proto=Onvif
          roles:
            - detect
        - path: rtsp://admin:xxx@192.168.0.209/cam/realmonitor?channel=1&subtype=00&authbasic=YWRtaW46TDI4NUYxNjQ=
          roles:
            - record
      hwaccel_args: -c:v h264_qsv
    rtmp:
      enabled: false
    detect:
      width: 640
      height: 480
      fps: 15
    objects:
      track:
        - person
        - bicycle
        - dog
        - cat
        - car
    snapshots:
      enabled: true
      bounding_box: true
  Giardino:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/Giardino_main?video=copy&audio=aac
          input_args: preset-rtsp-restream
          roles:
            - record
        - path: rtsp://127.0.0.1:8554/Giardino_sub?video=copy&audio=aac
          input_args: preset-rtsp-restream
          roles:
            - detect
      output_args:
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
    detect:
       width: 1536
       height: 576
       fps: 15
    motion:
      mask:
        - 0,0,388,0,0,504
        - 1536,0,1536,322,1384,440,1067,0
    objects:
      track:
        - person
        - dog
        - cat
        - car
    snapshots:
      enabled: true
      bounding_box: true
  Sala:
    ffmpeg:
      inputs:
        - path: rtsp://admin:xxx@192.168.0.64/Streaming/Channels/2
          roles:
            - detect
        - path: rtsp://admin:xxx@192.168.0.64/HighResolutionVideo
          roles:
            - record
      hwaccel_args: -c:v h264_qsv
    rtmp:
      enabled: false
    detect:
      width: 352
      height: 240
      fps: 15
    objects:
      track:
        - person
        - dog
        - cat
    snapshots:
      enabled: true
      bounding_box: true
birdseye:
  enabled: True
  mode: continuous
detectors:
  coral:
    type: edgetpu
    device: usb
record:
  enabled: True
  events:
    retain:
      default: 10
database:
  path: /media/frigate/db/frigate.db
```

### Relevant log output

```shell
2024-04-22 13:30:55.040393987  [INFO] Preparing Frigate...
2024-04-22 13:30:55.146665175  [INFO] Starting Frigate...
2024-04-22 13:30:59.418192086  [2024-04-22 13:30:59] frigate.app                    INFO    : Starting Frigate (0.13.2-6476f8a)
2024-04-22 13:30:59.508296829  [2024-04-22 13:30:59] peewee_migrate.logs            INFO    : Starting migrations
2024-04-22 13:30:59.532337595  [2024-04-22 13:30:59] peewee_migrate.logs            INFO    : There is nothing to migrate
2024-04-22 13:30:59.532944279  [2024-04-22 13:30:59] frigate.app                    INFO    : Running database vacuum
2024-04-22 13:31:06.297069174  [2024-04-22 13:31:06] frigate.app                    INFO    : Recording process started: 298
2024-04-22 13:31:06.300139864  [2024-04-22 13:31:06] frigate.app                    INFO    : go2rtc process pid: 87
2024-04-22 13:31:06.330572640  [2024-04-22 13:31:06] detector.coral                 INFO    : Starting detection process: 307
2024-04-22 13:31:06.337309636  [2024-04-22 13:31:06] frigate.app                    INFO    : Output process started: 309
2024-04-22 13:31:06.358061870  [2024-04-22 13:31:06] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as usb
2024-04-22 13:31:06.736953626  [2024-04-22 13:31:06] frigate.app                    INFO    : Camera processor started for Ingresso: 356
2024-04-22 13:31:06.744081631  [2024-04-22 13:31:06] frigate.app                    INFO    : Camera processor started for Garage: 361
2024-04-22 13:31:06.757147451  [2024-04-22 13:31:06] frigate.app                    INFO    : Camera processor started for Giardino: 363
2024-04-22 13:31:06.759484790  [2024-04-22 13:31:06] frigate.app                    INFO    : Camera processor started for Sala: 365
2024-04-22 13:31:06.767617607  [2024-04-22 13:31:06] frigate.app                    INFO    : Capture process started for Ingresso: 368
2024-04-22 13:31:06.776498209  [2024-04-22 13:31:06] frigate.app                    INFO    : Capture process started for Garage: 371
2024-04-22 13:31:06.786937042  [2024-04-22 13:31:06] frigate.app                    INFO    : Capture process started for Giardino: 375
2024-04-22 13:31:06.798431016  [2024-04-22 13:31:06] frigate.app                    INFO    : Capture process started for Sala: 383
2024-04-22 13:31:09.042802525  [2024-04-22 13:31:09] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
```

### FFprobe output from your camera

```shell
?
```

### Operating system

Proxmox

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

Hikvision, Reolink, D-link

**Comment by NickM-27** [comment]
We'd need to see what the contents of the folder actually are. 

Also, your hwaccel args should not be manual args but rather a preset

**Comment by Pinez83** [comment]
> We'd need to see what the contents of the folder actually are.
> 
> Also, your hwaccel args should not be manual args but rather a preset

Thank you for the reply. I admit that I am quite ignorant, so I ask you some questions related to what you say:
1) I don't know how to see the contents of the folder, in the LXC of proxmox, searching through the Console, the /tmp folder exists, but it doesn't have the /cache folder inside it... while in portainer I don't know where to look.
2) regarding the hwaccel args, I confess that I copied that part of the configuration online without understanding it, and then left it like that since everything seems to work. What do you recommend me to do?

**Comment by NickM-27** [comment]
1. You need to get a shell in the frigate container to view it
2. I'd suggest using preset-vaapi https://docs.frigate.video/configuration/hardware_acceleration/#via-vaapi

**Reply by Pinez83** [reply]
1. I can launch a Shell on proxmox, but I only see portainer (which contains frigate inside).  How can I throw a Shell inside portainer to see inside frigate?
2. I'll try immediately and let you know

**Reply by NickM-27** [reply]
1. inside the portainer lxc run `docker exec -it /bin/bash frigate`

**Comment by Pinez83** [comment]
With
docker exec -it frigate /bin/bash
I was able to see the tmp folder. Inside I have the following files
Garage@20240426185005+0000.mp4 Ingresso@20240426185004+0000.mp4
Giardino@20240426184954+0000.mp4 Sala@20240426185010+0000.mp4

the first word of each title is the name of one of the 4 cameras

**Reply by NickM-27** [reply]
that's all the files though? what are the sizes?

**Comment by Pinez83** [comment]
> that's all the files though? what are the sizes?

I noticed that the files change every 5-6 seconds, for example
Garage@20240429151140+0000.mp4  Garage@20240429151152+0000.mp4  Giardino@20240429151126+0000.mp4  Ingresso@20240429151146+0000.mp4

and 5 second later 
Garage@20240429151140+0000.mp4  Giardino@20240429151126+0000.mp4  Ingresso@20240429151156+0000.mp4
Garage@20240429151152+0000.mp4  Ingresso@20240429151146+0000.mp4

I have no idea how to understand the file sizes and how I can see if there is anything else, unfortunately I only know the "ls" command

**Reply by NickM-27** [reply]
you can run `ls -a`

but really this needs to be run when you are having issues. Right now it is emptying correctly

**Comment by Pinez83** [comment]
Ok...so why does the folder continue to fill up and is it always full even though I increase the size of the dedicated space?

**Reply by NickM-27** [reply]
what makes you think it is full? it looks like there are just a few files in there based on what you showed above

**Comment by Pinez83** [comment]
> what makes you think it is full? it looks like there are just a few files in there based on what you showed above

As you can see, on proxmox the space occupancy is more than 90%.  A month ago the space I had granted was only 6 GB and I had the same problem, so I increased but the problem returned
![Screenshot_20240429-173609](https://github.com/blakeblackshear/frigate/assets/96543064/20683d62-fe44-4485-b0d7-f63b6532878f)

inside the storage section on frigate, then, I saw that the problem is in the tmp folder...that's why I came to that deduction
![Screenshot_20240429-173654](https://github.com/blakeblackshear/frigate/assets/96543064/5af9564d-1e82-4ea3-8588-c46cf8c54e3c)

**Reply by NickM-27** [reply]
you should run `du -h /tmp/cache/*` and `df -h` and paste those here

**Comment by Pinez83** [comment]
> you should run `du -h /tmp/cache/*` and `df -h` and paste those here

**du -h /tmp/cache/***
260K    /tmp/cache/Garage@20240429155531+0000.mp4
4.8M    /tmp/cache/Giardino@20240429155526+0000.mp4
5.1M    /tmp/cache/Ingresso@20240429155526+0000.mp4
1.3M    /tmp/cache/Ingresso@20240429155536+0000.mp4

**df -h**
Filesystem                         Size  Used Avail Use% Mounted on
overlay                             12G   12G  133M  99% /
tmpfs                               64M     0   64M   0% /dev
shm                                1.0G   14M 1011M   2% /dev/shm
/dev/mapper/pve-vm--107--disk--0    12G   12G  133M  99% /config
//192.168.0.44/Frigate/clips       6.4T  177G  6.2T   3% /media/frigate/clips
//192.168.0.44/Frigate/recordings  6.4T  177G  6.2T   3% /media/frigate/recordings
none                               492K  4.0K  488K   1% /dev/bus/usb
udev                               3.8G     0  3.8G   0% /dev/bus/usb/002

**Reply by NickM-27** [reply]
it looks like you don't have a /tmp/cache mount to memory like it should be

**Reply by Pinez83** [reply]
do you mean on Portainer?

**Reply by NickM-27** [reply]
yes

**Comment by Pinez83** [comment]
> yes

I just checked the volumes mapped by frigate on portainer and in fact it isn't there.  How can I do to resolve it?
![Screenshot_20240430-095658](https://github.com/blakeblackshear/frigate/assets/96543064/738168f8-02fb-4d7c-8c6e-e37149cef44a)

**Reply by NickM-27** [reply, marked as answer]
https://docs.frigate.video/frigate/installation#storage

it is shown in the docs, a tmpfs volume should be added
